oxide-auth-rocket/src/lib.rs

```diff
@@ -131,49 +132,85 @@ impl<'r> WebRequest for OAuthRequest<'r> {
     }
 }
 
-impl<'r> WebResponse for Response<'r> {
+impl<'a, 'r> FromRequest<'a, 'r> for OAuthRequest<'r> {
+    type Error = NoError;
+
+    fn from_request(request: &'a Request<'r>) -> Outcome<Self, (Status, Self::Error), ()> {
+        Outcome::Success(Self::new(request))
+    }
+}
+
+/// Response type for Rocket OAuth requests
+///
+/// A simple wrapper type around a simple `rocket::Response<'r>` that implements `WebResponse`.
+#[derive(Debug)]
+pub struct OAuthResponse<'r>(pub Response<'r>);
```

Comment:
Preferred ordering for closely related components is

1. All struct definitions
2. Inherent methods and core trait impls
3. 'Helper' traits such as `Default`, `From`, ...

Please move the struct definition to below `OAuthRequest` and the `Default` and `From` impl under `WebResponse`.